src/Middleware/ErrorCatcher.php

```diff
@@ -123,6 +126,9 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
         try {
             return $handler->handle($request);
         } catch (Throwable $t) {
+            if ($this->eventDispatcher !== null) {
+                $this->eventDispatcher->dispatch(new ApplicationError($t));
+            }
```

Comment:
```suggestion
            try {
                if ($this->eventDispatcher !== null) {
                    $this->eventDispatcher->dispatch(new ApplicationError($t));
                }
            } catch (Throwable $t) {
                return $this->generateErrorResponse($t, $request);
            }
```

Reply:
What is it for?

Reply:
To handle exception correct that may be throws on dispatch `ApplicationError` event.